include/arkode/arkode_arkstep.h

```diff
@@ -124,146 +138,198 @@ SUNDIALS_EXPORT int ARKStepSetTableNum(void* arkode_mem,
-                                         SUNOutputFormat fmt);
-SUNDIALS_EXPORT char* ARKStepGetReturnFlagName(long int flag);
-
-SUNDIALS_EXPORT int ARKStepWriteParameters(void* arkode_mem, FILE* fp);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetEstLocalErrors instead")
+int ARKStepGetEstLocalErrors(void* arkode_mem, N_Vector ele);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetWorkSpace instead")
+int ARKStepGetWorkSpace(void* arkode_mem, long int* lenrw, long int* leniw);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetNumSteps instead")
+int ARKStepGetNumSteps(void* arkode_mem, long int* nsteps);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetActualInitStep instead")
+int ARKStepGetActualInitStep(void* arkode_mem, sunrealtype* hinused);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetLastStep instead")
+int ARKStepGetLastStep(void* arkode_mem, sunrealtype* hlast);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetCurrentStep instead")
+int ARKStepGetCurrentStep(void* arkode_mem, sunrealtype* hcur);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetCurrentTime instead")
+int ARKStepGetCurrentTime(void* arkode_mem, sunrealtype* tcur);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetCurrentState instead")
+int ARKStepGetCurrentState(void* arkode_mem, N_Vector* state);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetCurrentGamma instead")
+int ARKStepGetCurrentGamma(void* arkode_mem, sunrealtype* gamma);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetCurrentMassMatrix instead")
+int ARKStepGetCurrentMassMatrix(void* arkode_mem, SUNMatrix* M);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetTolScaleFactor instead")
+int ARKStepGetTolScaleFactor(void* arkode_mem, sunrealtype* tolsfac);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetErrWeights instead")
+int ARKStepGetErrWeights(void* arkode_mem, N_Vector eweight);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetResWeights instead")
+int ARKStepGetResWeights(void* arkode_mem, N_Vector rweight);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetNumGEvals instead")
+int ARKStepGetNumGEvals(void* arkode_mem, long int* ngevals);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetRootInfo instead")
+int ARKStepGetRootInfo(void* arkode_mem, int* rootsfound);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetNumConstrFails instead")
+int ARKStepGetNumConstrFails(void* arkode_mem, long int* nconstrfails);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetUserData instead")
+int ARKStepGetUserData(void* arkode_mem, void** user_data);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodePrintAllStats instead")
+int ARKStepPrintAllStats(void* arkode_mem, FILE* outfile, SUNOutputFormat fmt);
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeGetReturnFlagName instead")
+char* ARKStepGetReturnFlagName(long int flag);
+
+SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeWriteParameters instead")
+int ARKStepWriteParameters(void* arkode_mem, FILE* fp);
 
 SUNDIALS_EXPORT int ARKStepWriteButcher(void* arkode_mem, FILE* fp);
```

Comment:
I'd like to retain `ARKStepGetCurrentButcherTables` in case people wanted to use the table for some other purpose.  That said, I agree that there are other routines to write out the Butcher tables, so I would be fine with deprecating `ARKStepWriteButcher`.